### PR TITLE
Converted SGA into django app and added tox base testing

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
fixes https://github.com/mitodl/edx-sga/issues/171

#### What's this PR do?
It converts SGA to django app, and adds tox base testing for different versions of django

#### How should this be manually tested?
go to app, install requirement and run tox.

### CODE REVIEW

**Comment by pdpinch** [comment]
you've got a typo in the title

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[tox]
+envlist = py{27}-django{18,19,110,111}, quality
```
Comment (noisecapella) [inline tox.ini]: What about Python 3?

**Comment by noisecapella** [comment]
I'm getting these errors in the `quality` tests:

    edx_sga/integration_tests.py:74:80: E501 line too long (82 > 79 characters)
    edx_sga/integration_tests.py:107:80: E501 line too long (110 > 79 characters)
    edx_sga/integration_tests.py:664:80: E501 line too long (86 > 79 characters)
    edx_sga/sga.py:225:80: E501 line too long (85 > 79 characters)
    edx_sga/sga.py:538:80: E501 line too long (87 > 79 characters)
    edx_sga/sga.py:544:80: E501 line too long (83 > 79 characters)
    edx_sga/sga.py:566:80: E501 line too long (86 > 79 characters)
    edx_sga/sga.py:628:80: E501 line too long (96 > 79 characters)

File: edx_sga/settings.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+"""
```
Comment (noisecapella) [inline edx_sga/settings.py]: Can you rename this file to `test_settings.py` to make it clear that these settings are never meant for use in production?

File: edx_sga/settings.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+Django settings for edx_sga_t project.
```
Comment (noisecapella) [inline edx_sga/settings.py]: Please change this to `edx_sga`

**Comment by noisecapella** [comment]
Can you add a `pytest.ini` file with the `.tox` directory in `norecursedirs`?

File: test_requirements.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+ddt==1.1.1
+coverage==4.0.3
+edx-lint==0.4.0
+mock==1.3.0
+pep8==1.6.2
+pytz
+codecov
+tox>=2.3.1,<3.0.0
```
Comment (noisecapella) [inline test_requirements.txt]: Why not a newer version of tox?

File: edx_sga/tests.py [hunk 5]
```diff
@@ -1,676 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (noisecapella) [inline edx_sga/tests.py]: It looks like this file still exists. Can you delete it instead of just clearing its contents?

**Comment by amir-qayyum-khan** [comment]
@noisecapella please review

File: tox.ini (hunk 1 above)
Comment (amir-qayyum-khan) [inline tox.ini]: edX platform do not support python 3

File: README.rst [hunk 6]
```diff
@@ -252,11 +252,14 @@ Staff Grading
...
+
+    $ python manage.py lms --settings=test test edx_sga.integration_tests
 
 To get statement coverage::
 
-    $ coverage run --source edx_sga manage.py lms --settings=test test edx_sga
+    $ coverage run --source edx_sga manage.py lms --settings=test test edx_sga.integration_tests
     $ coverage report -m
+
+To run tests on different versions of django with mocking edX platform use:
```
Comment (noisecapella) [inline README.rst]: Please add an extra `:` to the end of this line, and add a blank line after this line.

File: edx_sga/test_settings.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    'django.middleware.common.CommonMiddleware',
+    'django.middleware.csrf.CsrfViewMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+    'django.middleware.security.SecurityMiddleware',
+)
+
+# ROOT_URLCONF = 'edx_sga.urls'
```
Comment (noisecapella) [inline edx_sga/test_settings.py]: Can you remove this line if it's not necessary?

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella

**Pull request merged**